nvidia-bert/ort_addon/run_pretraining_ort.py

```diff
@@ -32,12 +32,21 @@
 import os
 import numpy as np
 import torch
+import torch.distributed as dist
 from torch.utils.data import DataLoader, RandomSampler, SequentialSampler, Dataset
 import math
 import multiprocessing
 import modeling
 
-from utils import is_main_process, format_step
+from utils import format_step
+
+# replace routine from utils.py as we dont use torch.distributed
+def is_main_process(args):
```

Comment:
i think we installed mpi4py in the image. can we just use that?

Reply:
There's a setup routine in ort_supplement that either gets the rank from mpi4py or from environment variables in AML. In either case it stores the rank in args. So that was convenient to capture either case.

Reply:
It's possible the AML route was fine because torch.distibuted.init() is called and I think it might be possible that snoops the process arguments. I'm not totally sure though.